### ISSUE TITLE
Replacing favicon.ico not working for NextJS new version

### ISSUE THREAD

**Issue by linghuccc** [vercel/next.js#64597]
### Link to the code that reproduces this issue

https://codesandbox.io/p/github/linghuccc/nextJs/main

### To Reproduce

Step 1: Install project using `npx create-next-app@latest`

Step 2: Replace file `favicon.ico` under `src/app/`

### Current vs. Expected behavior

I used to replace favicon.ico file to change the web icon for the page. But recently it does not seem to be working.

### Provide environment information

```bash
Operating System:
	Platform: Ubuntu
	Version: Ubuntu 22.04.2 LTS

Binaries:
	Node: v18.18.2
	npm: 10.5.0

Relevant Packages:
	next: 14.2.1
	react: 18
	react-dom: 18
	typescript: 5
```

### Which area(s) are affected? (Select all that apply)

Not sure

### Which stage(s) are affected? (Select all that apply)

next dev (local)

### Additional context

One thing I can not explain is that the old projects used to work now also failed to work. I really do not know how to deal this situation.

**Comment by jenewland1999** [comment]
@linghuccc Is it possible the browser is caching the Favicon for a short period of time and so the new one doesn't show up because of that?

I've seen this sort of behaviour now and again.

**Comment by samcx** [comment]
@linghuccc I believe @jenewland1999 is right here! I am not seeing this issue. To prove the point above, I also saw the Vercel `favicon.ico` first (since I've already started up a `create-next-app` once), but after a couple browser refreshes (Arc browser), I am seeing the correct `favicon.ico`.

![CleanShot 2024-04-17 at 14 49 03@2x](https://github.com/vercel/next.js/assets/28912696/714284bd-534c-4064-8e73-6d164243b228)
